packages/frontend/core/src/bootstrap/global-error-handler.ts

```diff
@@ -0,0 +1,46 @@
+import errorImg from './error.png';
+
+function handleGlobalUnrecoverableError(e: ErrorEvent | PromiseRejectionEvent) {
+  e.stopImmediatePropagation();
+  const div = document.createElement('div');
+  div.innerHTML = `<style>
+.gue {display:flex;flex-direction:column;align-items:center;justify-content:center;width:380px;}
+.gue img{width:380px;}
+.gue div{padding:16px 40px 0 40px;text-align:center;}
+.gue .p1{color:#141414;line-height:24px;font-weight:500;}
+.gue .p2{color:#7A7A7A;line-height:22px;}
+</style>
+<div class="gue">
+  <img src="${errorImg}" />
+  <div>
+    <p class="p1">Unsupported Environment</p>
+    <p class="p2">
+      It looks like AFFiNE cannot run in this environment.
+      Please ensure you are using a supported browser or update your device's operating system to the latest version.
+      If the issue persists, visit our <a href="https://github.com/toeverything/AFFiNE/issues">support page</a> for further assistance.
+    </p>
+  </div>
+</div>`;
+  div.setAttribute(
+    'style',
+    'position:absolute;top:0;left:0;height:100vh;width:100vw;display:flex;flex-direction:column;align-items:center;justify-content:center;background:white;z-index:999;'
+  );
+  document.body.append(div);
+  if ('error' in e) {
+    console.error(e.error.stack);
+  } else {
+    console.error(e.reason.stack);
+  }
+}
+
+function registerGlobalErrorHandler() {
+  if (typeof document !== 'undefined') {
+    globalThis.addEventListener(
+      'unhandledrejection',
+      handleGlobalUnrecoverableError
+    );
+    globalThis.addEventListener('error', handleGlobalUnrecoverableError);
+  }
+}
+
+registerGlobalErrorHandler();
```

Comment:
If the browser version is too low, there is unparseable syntax in the js file, and I suspect this will not be run at all.

should we bundle the error handling into a separate js file ?

Reply:
which one

Reply:
it won't help but more complex. I will split this into a separated chunk to make sure it will always be loaded first